### PR TITLE
Provide dlang-community repo

### PR DESCRIPTION
**Pull request by stefan-koch-sociomantic**
Same as in title.
Adds a configuarion file for the dlang-community apt repo;
and installs dfmt from it

### CODE REVIEW

**Comment by nemanja-boric-sociomantic** [comment]
The second commit should say: "Install dfmt version 0.6.* in dlang image".

**Comment by nemanja-boric-sociomantic** [comment]
Other than that LGTM

**Comment by stefan-koch-sociomantic** [comment]
@nemanja-boric-sociomantic changed

**Review by nemanja-boric-sociomantic — APPROVED** [review]

File: docker/base [hunk 1]
```diff
@@ -25,11 +25,18 @@ apt -y install apt-transport-https build-essential bzip2 devscripts \
...
 git config --system user.email "tsunami@sociomantic.com"
 
-# Add extra APT repositories from Sociomantic in Bintray
+# Add extra APT repository from dlang-community in Bintray
+dist="$(lsb_release -cs)"
 apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 379CE192D401AB61
+
+echo '# dlang-community repos' \
+        > /etc/apt/sources.list.d/dlang-community.list
+echo "deb https://dl.bintray.com/dlang-community/apt $dist prerelease release" \
```
Comment (leandro-lucarella-sociomantic) [inline docker/base]: The prerelease shouldn't be included for now, as we don't have `apt-preferences` properly configured not to pick the prereleases automatically. We should add prereleases at some point, but we need to make sure that they are properly configured.

**Review by leandro-lucarella-sociomantic — CHANGES_REQUESTED** [review]
Please also add simple tests to check if `dfmt` is really installed and the package doesn't get unintentionally upgraded, have a look at:

* https://github.com/sociomantic-tsunami/cachalot/blob/v3.x.x/test/dlang-user
* https://github.com/sociomantic-tsunami/cachalot/blob/v3.x.x/test/dlang-root

**Comment by leandro-lucarella-sociomantic** [comment]
And Travis is not happy, I guess the package is not really in the repos yet?
https://travis-ci.org/sociomantic-tsunami/cachalot/builds/291593128#L2034

**Comment by leandro-lucarella-sociomantic** [comment]
**SIDE NOTE:** I'm actually considering adding minor verions to cachalot ( ping @mihails-strasuns ), I think putting this in `v4` might be overkill, and making it a patch release also sounds weird. I will have to check if the infra behind this is prepared for minor version releases though.

**Comment by mihails-strasuns-sociomantic** [comment]
Hm, I don't see any practical reason to go for minors right now. I mean the only reason why we have any versions at all (instead of always using `sociomantictsunami/dlang:latest`) is to avoid silent unexpected upgrades of default packages. It has not direct relation with versioning of those packages and always acts as whole snapshot.

`v4` sounds just fine. Only problem is that if you try to use neptune branch model for it, you will have too many branches at the same time.

**Comment by stefan-koch-sociomantic** [comment]
I have no idea why this fails actually, the package is in the repo and if I execute the added commands manually in the docker-contrainer it works.

**Comment by leandro-lucarella-sociomantic** [comment]
> Hm, I don't see any practical reason to go for minors right now. I mean the only reason why we have any versions at all (instead of always using sociomantictsunami/dlang:latest) is to avoid silent unexpected upgrades of default packages. It has not direct relation with versioning of those packages and always acts as whole snapshot.

Well, not entirely, the intention was never really to use `:latest`, as if that's done old branches could stop building if there is a new patch release and the cachalot image changes. The idea is to always use explicit versions. Also, since we now pin versions of the important (external) programs (the internal ubuntu programs are supposed to also only receive bugfix updates), adding a new program doesn't really look like could break anything in the current version (is similar to add symbols in a library, the theoretical possibility to break something exists but is so small that it doesn't make to consider it a breaking change).

**Comment by leandro-lucarella-sociomantic** [comment]
> v4 sounds just fine. Only problem is that if you try to use neptune branch model for it, you will have too many branches at the same time.

I wouldn't go for a full neptune scheme either, since users are expected to only pin on majors. Or put it another way, I would go for full neptune but I will only keep updating only the latest minor. From the practical experience, there were barely any update to images.

**Comment by stefan-koch-sociomantic** [comment]
@leandro-lucarella-sociomantic for now I only added the dlang-community repo this should make the pr pass.

**Comment by leandro-lucarella-sociomantic** [comment]
@stefan-koch-sociomantic https://github.com/sociomantic-tsunami/cachalot/pull/37#pullrequestreview-71247107 still needs to be addressed :)

**Comment by stefan-koch-sociomantic** [comment]
@leandro-lucarella-sociomantic since this pr does not install dfmt it cannot check it.
I will keep it in mind though

**Comment by leandro-lucarella-sociomantic** [comment]
Ah, right. But then why isn't provided, is just for initial testing? Because I think cachalot should provide dfmt.

**Comment by stefan-koch-sociomantic** [comment]
@leandro-lucarella-sociomantic I'd like to do it in two pr's so they'll have a change to all pass testing.

**Comment by nemanja-boric-sociomantic** [comment]
But you don't know if this one is passing, as your changes are not actually tested.

**Comment by stefan-koch-sociomantic** [comment]
As a side-note: should we really do all the dlang-specific apt-setup in the base image rather then the dlang image ?

**Comment by leandro-lucarella-sociomantic** [comment]
> As a side-note: should we really do all the dlang-specific apt-setup in the base image rather then the dlang image ?

No, good catch, the sociomantic bintray repo is there because we have tools there that are not dlang specific (like `git-hub`), but the dlang-specific stuff should go to the dlang image.

**Comment by stefan-koch-sociomantic** [comment]
> No, good catch, the sociomantic bintray repo is there because we have tools there that are not dlang specific (like git-hub), but the dlang-specific stuff should go to the dlang image.

pr is in progress.

**Comment by leandro-lucarella-sociomantic** [comment]
Blocked on #38.

**Comment by stefan-koch-sociomantic** [comment]
@leandro-lucarella-sociomantic just rebased on v4

**Review by leandro-lucarella-sociomantic — APPROVED** [review]

**Comment by leandro-lucarella-sociomantic** [comment]
Thanks!

**Pull request merged**